lib/ansible/modules/cloud/google/gcp_compute_disk.py

```diff
@@ -450,8 +456,44 @@ def create(module, link, kind):
     return wait_for_operation(module, auth.post(link, resource_to_request(module)))
 
 
-def update(module, link, kind):
-    module.fail_json(msg="Disk cannot be edited")
+def update(module, link, kind, fetch):
+    update_fields(module, resource_to_request(module),
+                  response_to_hash(module, fetch))
+    return fetch_resource(module, self_link(module), kind)
+
+
+def update_fields(module, request, response):
+    if response.get('labels') != request.get('labels'):
+        labelfingerprint_update(module, request, response)
+    if response.get('sizeGb') != request.get('sizeGb'):
+        sizegb_update(module, request, response)
+
+
+def labelfingerprint_update(module, request, response):
```

Comment:
It would help readability for this to be `label_fingerprint_update`.

Reply:
Good call. Fixed! Running through the magician now.